### PR TITLE
Fetching posts

### PR DESCRIPTION
**Pull request by atwfk**
- [x] install `@m3ntorship/posts-client`
- [x] created api baseURL
- [x] created `getPosts` function for fetching posts
- [x] install `axios` for declaring `getPosts` type
- [x] fetching posts with `getServerSideProps`

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -10,8 +13,22 @@ export default function Home(): ReactElement {
...
+          <p>{postsCount}</p>
         </div>
       </div>
     </>
   );
-}
+};
+
+export const getServerSideProps: GetServerSideProps = async () => {
+  const { data } = await getPosts();
```
Comment (Nilegfx) [inline pages/index.tsx:24]: error handling, PLEASE.

**Comment by atwfk** [comment]
@Nilegfx requested changes have been updated

File: pages/index.tsx [hunk 2]
```diff
@@ -21,7 +30,18 @@ const Home: FC<InlineResponse200> = ({ postsCount }): ReactElement => {
 };
 
 export const getServerSideProps: GetServerSideProps = async () => {
-  const { data } = await getPosts();
+  const { message, response } = (await getPosts()) as IErrorHandle.IErrorData;
```
Comment (Nilegfx) [inline pages/index.tsx:33]: WHAT IS THIS!

File: pages/index.tsx [hunk 3]
```diff
@@ -21,7 +30,18 @@ const Home: FC<InlineResponse200> = ({ postsCount }): ReactElement => {
...
+  if (message) {
+    return {
+      props: {
+        message,
+        statusCode: response.status,
+      },
+    };
+  }
+
+  const { data } = (await getPosts()) as AxiosResponse<InlineResponse200>;
```
Comment (Nilegfx) [inline pages/index.tsx:44]: I don't understand what's going on here.

File: pages/index.tsx (hunk 2 above)
Comment (atwfk) [reply pages/index.tsx:33]: here i'm destructing `message` and `response` if we have an error and passing them to `<Error  />` component that imported from `'next/error'`

**Review by Nilegfx — CHANGES_REQUESTED** [review]
DON'T MERGE THIS.

File: pages/index.tsx (hunk 2 above)
Comment (Nilegfx) [reply pages/index.tsx:33]: where did you read this approach?

Comment (atwfk) [reply pages/index.tsx:33]: no where, there wasn't many resources for error handling in `getServerSideProps`, i want more explanation for what is going on wrong and i will fix it

Comment (Nilegfx) [reply pages/index.tsx:33]: I'll join the stand up. let's talk there.

File: modules/shared/api/postsApi.api.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { PostsApi } from '@m3ntorship/posts-client';
+import type { Configuration } from '@m3ntorship/posts-client/dist/client';
+
+export const postsApi = new PostsApi({
+  basePath: 'http://localhost:4010',
```
Comment (Nilegfx) [inline modules/shared/api/postsApi.api.ts:5]: for niw, this is ok. in the future, this should be a configuration not hard-coded

File: modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.test.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      postTitle: 'a7la mesa 3lek!',
+    };
+    const mockComponent = jest.fn(function <T extends IDataTests.IDataSucccess>(
+      data: T,
+    ) {
+      return <div>{data.postTitle}</div>;
+    });
+    const Component = WithErrorHandler(mockComponent);
+    render(<Component data={successData} />);
+    expect(mockComponent).toBeCalled();
```
Comment (Nilegfx) [inline modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.test.tsx:18]: I would assert that the final data is displayed, not the component has been called.

File: modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.test.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import WithErrorHandler from './WithErrorHandler';
+import type { IDataTests } from './IDataTests';
+
```
Comment (Nilegfx) [inline modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.test.tsx:5]: I believe one test that renders the same component twice (one with success data and another with error) should be the way to test this HoC.

File: modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.tsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import type { ReactElement, FC } from 'react';
+import type { IWithErrorHandler } from './IwithErrorHandler';
+
+function withErrorHandler<T>(WrappedComponent: FC<T>): FC<T> {
+  const Result: FC<IWithErrorHandler.IProps & T> = (props): ReactElement => {
```
Comment (Nilegfx) [inline modules/shared/components/HOC/WithErrorHandler/WithErrorHandler.tsx:6]: `result` is not a descriptive name

File: modules/shared/types/postFeed/IPostFeed.d.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import type { EPostType } from './EPostType';
```
Comment (Nilegfx) [inline modules/shared/types/postFeed/IPostFeed.d.ts:1]: I remember we talked about writing FE types or use the client Types, can someone explain to me why can't we use the client types? it is not a blocker for this PR. I am just asking for better understanding

**Review by Nilegfx — COMMENTED** [review]
some minor comments.

**Pull request merged**